python/ray/experimental/data/dataset.py

```diff
@@ -465,14 +588,41 @@ def to_tf(self, **todo) -> "ray.util.sgd.tf.TFMLDataset":
         raise NotImplementedError  # P1
 
     def to_dask(self) -> "dask.DataFrame":
-        """Convert this dataset into a Dask dataframe.
+        """Convert this dataset into a Dask DataFrame.
 
         Time complexity: O(1)
 
         Returns:
-            A Dask dataframe created from this dataset.
+            A Dask DataFrame created from this dataset.
         """
-        raise NotImplementedError  # P1
+        import dask
+        import dask.dataframe as dd
+        from ray.util.client.common import ClientObjectRef
+
+        def _block_to_df(block):
+            return block._table.to_pandas()
+
+        _remote_block_to_df = ray.remote(_block_to_df)
+
+        @dask.delayed
+        def block_to_df(block):
+            if isinstance(block, (ray.ObjectRef, ClientObjectRef)):
+                # The block Ray future hasn't been resolved to a concrete
+                # value, so the end-user isn't using Dask-on-Ray. We therefore
```

Comment:
We shouldn't support non-dask-on-ray here.

Reply:
Is there a reason why? It's very easy to support here.

Reply:
It's confusing--- users will definitely accidentally not use it by accident even when they mean to.

Reply:
I suppose that we can raise an error here if the Ray futures haven't been resolved.